components/org.wso2.carbon.metrics.manager/src/main/java/org/wso2/carbon/metrics/manager/MetricHierarchy.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package org.wso2.carbon.metrics.manager;
+
+import java.util.concurrent.ConcurrentMap;
+
+public interface MetricHierarchy extends Iterable<MetricHierarchy> {
+    /**
+     * @return The level (depth) of the current node.
+     */
+    int getLevel();
+
+    /**
+     * @return The map of available {@link Metric}s for current node.
+     */
+    ConcurrentMap<String, Metric> getMetrics();
```

Comment:
@grainier Better use the interface here instead of the concrete implementation as the type